tests/module_loading/module/module.c

```diff
@@ -37,3 +39,25 @@ int add_with_constant(volatile int a, volatile int b)
         t = -1;
     return t;
 }
+
+#define TEST_SYMBOL(name)                 \
+    do                                    \
+    {                                     \
+        volatile void* ptr = (void*)name; \
+        (void)ptr;                        \
+    } while (0)
+
+void test_libc_symbols()
+{
+    TEST_SYMBOL(memcmp);
```

Comment:
The contract with symcrypt also has the constraint that `sizeof(oe_mutex_t) <= sizeof(GLIBC's pthread_mutex_t)`.
It would be good to lock this down here so that
- We don't regress in the future
- The contract is complete

Reply:
Update the PR to use musl headers directly and checks the size of `pthread_mutex_t`